src/common/shutdownHandler.ts

```diff
@@ -0,0 +1,28 @@
+const NOT_FOUND_INDEX = -1;
+
+export type ShutdownFunc = () => Promise<void>;
+
+export class ShutdownHandler {
```

Comment:
should be a singleton so on each resolve will return the same instance

Reply:
its already singleton because its useValue